Yoast/javascript
# Add a check for keywords containing a dash

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds a check for keywords containing a dash as part of the word (e.g., buku-buku) so that they can be found in the slug.
* [Yoast SEO Free] Improves recognition of keywords that contain a hyphen in slug (for example: re-duplicated, on-the-go).

## Test instructions

This PR can be tested by following these steps:

* Check the `KeywordCountInUrlSpec` file in Javascript:
   - add a console.log in `KeywordCountInUrl` to print the outcome of the URL check in `KeywordCountInUrlSpec`. The 
         argument of the console.log is the function `keyphraseInSlug`;
   - run `KeywordCountInUrlSpec` and check that all tests pass;
   - in the test result also check that the last result (corresponding to the example buku-buku in the spec) prints this as a 
         result ( where `countWordMatches` is 1 as buku-buku is seen as one word and `percentWordMatches` is 100 as the 
          word was completely found.): 

`console.log src/researches/keywordCountInUrl.js:26
    { countWordMatches: 1,
      percentWordMatches: 100,
      keyphraseOrSynonym: 'keyphrase' }`

  - Add more specs of dashed words, even from different languages (e.g., re-duplication, causa-effetto), and see if they get recognised with right word-count and word-match. Delete them after testing.

* Test in the content-analysis app:
  - start the content-analysis app and run `yarn start`
  - Paste a text of at least 150 words
  - Set the locale to id_ID (or change it to the language you are testing in)
  - Switch the morphology button to Off
  - Set the keyphrase and the slug as `buku-buku` 
  - Check in the assessment that the keyword is found in the slug 
  - Check that the functionality still works when the keyphrase is buku ( and the slug also contains buku) 
  - Check that the keyphrase buku-buku is not matched when the slug is buku.

## QA test instructions 
Check that words containing a hyphen are found in the slug when used as keywords. Here is the expected behaviour:

* Buku-buku
   - Set the locale to Indonesian (id_ID)
   - Set `buku-buku` as the keyphrase and use `buku-buku` in the slug

* Re-duplication
   - Set the locale to English (en_EN)
   - Set `re-duplication` as the keyphrase and use `re-duplication` in the slug

* Two-room apartment
   - Set the locale to English (en_EN)
   - Set `two-room-apartment` as the keyphrase and use `two-room apartment` in the slug 
  

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-460
